### DISCUSSION TITLE
Home assistant pjsip_custom.conf error

### DISCUSSION THREAD

**Discussion by pilurizzu** [TECH7Fox/asterisk-hass-addons#197] [Q&A]
[8001]
type = friend
context = default
disallow = all
allow = alaw,ulaw          ; Audio codecs
allow=h264                 ; Video codecs
direct_media_method=invite
dtmf_mode=info
callerid="Doorbell" <8001>
force_rport=no
aors= 8001
auth = auth8001

[8001]
type = aor
max_contacts = 1

[auth8001]
type=auth
auth_type=userpass
password= 8001    
username= 8001

"ASTERISK REGISTER"
![Screenshot_20221215_184758_Samsung Internet](https://user-images.githubusercontent.com/76753636/207932338-272d2413-ccd5-433f-96d7-906e754f6f8a.png)

[Dec 15 18:42:01] DEBUG[495]: res_pjsip/pjsip_distributor.c:393 find_dialog: Could not find matching transaction for Request msg REGISTER/cseq=1 (rdata0x55b5312e58)
[Dec 15 18:42:01] DEBUG[495]: res_pjsip/pjsip_distributor.c:471 ast_sip_get_distributor_serializer: Calculated serializer pjsip/distributor-00000032 to use for Request msg REGISTER/cseq=1 (rdata0x55b5312e58)
[Dec 15 18:42:01] DEBUG[493]: threadpool.c:536 grow: Increasing threadpool pjsip/pool's size by 5
[Dec 15 18:42:01] DEBUG[567]: res_pjsip_endpoint_identifier_ip.c:275 common_identify: No identify sections to match against
[Dec 15 18:42:01] DEBUG[567]: res_pjsip_endpoint_identifier_user.c:148 username_identify: Attempting identify by From username '8001' domain '192.168.1.250'
[Dec 15 18:42:01] DEBUG[567]: res_pjsip_endpoint_identifier_user.c:152 username_identify: Endpoint not found for From username '8001' domain '192.168.1.250'
[Dec 15 18:42:01] DEBUG[567]: res_pjsip_endpoint_identifier_ip.c:275 common_identify: No identify sections to match against
[Dec 15 18:42:01] NOTICE[567]: res_pjsip/pjsip_distributor.c:676 log_failed_request: Request 'REGISTER' from '<sip:8001@192.168.1.250>' failed for '192.168.1.110:5060' (callid: 156578329@192.168.1.110) - No matching endpoint found
[Dec 15 18:42:01] SECURITY[563]: res_security_log.c:114 security_event_stasis_cb: SecurityEvent="InvalidAccountID",EventTV="2022-12-15T18:42:01.712+0100",Severity="Error",Service="PJSIP",EventVersion="1",AccountID="8001",SessionID="156578329@192.168.1.110",LocalAddress="IPV4/UDP/192.168.1.250/5060",RemoteAddress="IPV4/UDP/192.168.1.110/5060"
[Dec 15 18:42:01] DEBUG[567]: res_pjsip_authenticator_digest.c:358 verify: Realm: asterisk  Username:   Result: NOAUTH
[Dec 15 18:42:01] SECURITY[563]: res_security_log.c:114 security_event_stasis_cb: SecurityEvent="ChallengeSent",EventTV="2022-12-15T18:42:01.712+0100",Severity="Informational",Service="PJSIP",EventVersion="1",AccountID="<unknown>",SessionID="156578329@192.168.1.110",LocalAddress="IPV4/UDP/192.168.1.250/5060",RemoteAddress="IPV4/UDP/192.168.1.110/5060",Challenge=""
[Dec 15 18:42:02] DEBUG[495]: res_pjsip/pjsip_distributor.c:393 find_dialog: Could not find matching transaction for Request msg REGISTER/cseq=2 (rdata0x7f40012aa8)
[Dec 15 18:42:02] DEBUG[495]: res_pjsip/pjsip_distributor.c:471 ast_sip_get_distributor_serializer: Calculated serializer pjsip/distributor-00000032 to use for Request msg REGISTER/cseq=2 (rdata0x7f40012aa8)
[Dec 15 18:42:02] DEBUG[567]: res_pjsip_endpoint_identifier_ip.c:275 common_identify: No identify sections to match against
[Dec 15 18:42:02] DEBUG[567]: res_pjsip_endpoint_identifier_user.c:148 username_identify: Attempting identify by From username '8001' domain '192.168.1.250'
[Dec 15 18:42:02] DEBUG[567]: res_pjsip_endpoint_identifier_user.c:152 username_identify: Endpoint not found for From username '8001' domain '192.168.1.250'
[Dec 15 18:42:02] DEBUG[567]: res_pjsip_endpoint_identifier_ip.c:275 common_identify: No identify sections to match against
[Dec 15 18:42:02] NOTICE[567]: res_pjsip/pjsip_distributor.c:676 log_failed_request: Request 'REGISTER' from '<sip:8001@192.168.1.250>' failed for '192.168.1.110:5060' (callid: 156578329@192.168.1.110) - No matching endpoint found
[Dec 15 18:42:02] DEBUG[567]: res_pjsip_authenticator_digest.c:358 verify: Realm: asterisk  Username:   Result: NOAUTH
[Dec 15 18:42:02] SECURITY[563]: res_security_log.c:114 security_event_stasis_cb: SecurityEvent="InvalidAccountID",EventTV="2022-12-15T18:42:02.720+0100",Severity="Error",Service="PJSIP",EventVersion="1",AccountID="8001",SessionID="156578329@192.168.1.110",LocalAddress="IPV4/UDP/192.168.1.250/5060",RemoteAddress="IPV4/UDP/192.168.1.110/5060"
[Dec 15 18:42:02] SECURITY[563]: res_security_log.c:114 security_event_stasis_cb: SecurityEvent="ChallengeSent",EventTV="2022-12-15T18:42:02.720+0100",Severity="Informational",Service="PJSIP",EventVersion="1",AccountID="<unknown>",SessionID="156578329@192.168.1.110",LocalAddress="IPV4/UDP/192.168.1.250/5060",RemoteAddress="IPV4/UDP/192.168.1.110/5060",Challenge=""
[Dec 15 18:42:03] DEBUG[495]: res_pjsip/pjsip_distributor.c:393 find_dialog: Could not find matching transaction for Request msg REGISTER/cseq=3 (rdata0x7f400030e8)
[Dec 15 18:42:03] DEBUG[495]: res_pjsip/pjsip_distributor.c:471 ast_sip_get_distributor_serializer: Calculated serializer pjsip/distributor-00000032 to use for Request msg REGISTER/cseq=3 (rdata0x7f400030e8)
[Dec 15 18:42:03] DEBUG[567]: res_pjsip_endpoint_identifier_ip.c:275 common_identify: No identify sections to match against
[Dec 15 18:42:03] DEBUG[567]: res_pjsip_endpoint_identifier_user.c:148 username_identify: Attempting identify by From username '8001' domain '192.168.1.250'
[Dec 15 18:42:03] DEBUG[567]: res_pjsip_endpoint_identifier_user.c:152 username_identify: Endpoint not found for From username '8001' domain '192.168.1.250'
[Dec 15 18:42:03] DEBUG[567]: res_pjsip_endpoint_identifier_ip.c:275 common_identify: No identify sections to match against
[Dec 15 18:42:03] NOTICE[567]: res_pjsip/pjsip_distributor.c:676 log_failed_request: Request 'REGISTER' from '<sip:8001@192.168.1.250>' failed for '192.168.1.110:5060' (callid: 156578329@192.168.1.110) - No matching endpoint found
[Dec 15 18:42:03] SECURITY[563]: res_security_log.c:114 security_event_stasis_cb: SecurityEvent="InvalidAccountID",EventTV="2022-12-15T18:42:03.748+0100",Severity="Error",Service="PJSIP",EventVersion="1",AccountID="8001",SessionID="156578329@192.168.1.110",LocalAddress="IPV4/UDP/192.168.1.250/5060",RemoteAddress="IPV4/UDP/192.168.1.110/5060"
[Dec 15 18:42:03] DEBUG[567]: res_pjsip_authenticator_digest.c:358 verify: Realm: asterisk  Username:   Result: NOAUTH
[Dec 15 18:42:03] SECURITY[563]: res_security_log.c:114 security_event_stasis_cb: SecurityEvent="ChallengeSent",EventTV="2022-12-15T18:42:03.749+0100",Severity="Informational",Service="PJSIP",EventVersion="1",AccountID="<unknown>",SessionID="156578329@192.168.1.110",LocalAddress="IPV4/UDP/192.168.1.250/5060",RemoteAddress="IPV4/UDP/192.168.1.110/5060",Challenge=""
[Dec 15 18:42:04] DEBUG[495]: res_pjsip/pjsip_distributor.c:393 find_dialog: Could not find matching transaction for Request msg REGISTER/cseq=4 (rdata0x7f400024f8)
[Dec 15 18:42:04] DEBUG[495]: res_pjsip/pjsip_distributor.c:471 ast_sip_get_distributor_serializer: Calculated serializer pjsip/distributor-00000032 to use for Request msg REGISTER/cseq=4 (rdata0x7f400024f8)
[Dec 15 18:42:04] DEBUG[567]: res_pjsip_endpoint_identifier_ip.c:275 common_identify: No identify sections to match against
[Dec 15 18:42:04] DEBUG[567]: res_pjsip_endpoint_identifier_user.c:148 username_identify: Attempting identify by From username '8001' domain '192.168.1.250'
[Dec 15 18:42:04] DEBUG[567]: res_pjsip_endpoint_identifier_user.c:152 username_identify: Endpoint not found for From username '8001' domain '192.168.1.250'
[Dec 15 18:42:04] DEBUG[567]: res_pjsip_endpoint_identifier_ip.c:275 common_identify: No identify sections to match against
[Dec 15 18:42:04] NOTICE[567]: res_pjsip/pjsip_distributor.c:676 log_failed_request: Request 'REGISTER' from '<sip:8001@192.168.1.250>' failed for '192.168.1.110:5060' (callid: 156578329@192.168.1.110) - No matching endpoint found
[Dec 15 18:42:04] SECURITY[563]: res_security_log.c:114 security_event_stasis_cb: SecurityEvent="InvalidAccountID",EventTV="2022-12-15T18:42:04.795+0100",Severity="Error",Service="PJSIP",EventVersion="1",AccountID="8001",SessionID="156578329@192.168.1.110",LocalAddress="IPV4/UDP/192.168.1.250/5060",RemoteAddress="IPV4/UDP/192.168.1.110/5060"
[Dec 15 18:42:04] DEBUG[567]: res_pjsip_authenticator_digest.c:358 verify: Realm: asterisk  Username:   Result: NOAUTH
[Dec 15 18:42:04] SECURITY[563]: res_security_log.c:114 security_event_stasis_cb: SecurityEvent="ChallengeSent",EventTV="2022-12-15T18:42:04.795+0100",Severity="Informational",Service="PJSIP",EventVersion="1",AccountID="<unknown>",SessionID="156578329@192.168.1.110",LocalAddress="IPV4/UDP/192.168.1.250/5060",RemoteAddress="IPV4/UDP/192.168.1.110/5060",Challenge=""
[Dec 15 18:42:05] DEBUG[495]: res_pjsip/pjsip_distributor.c:393 find_dialog: Could not find matching transaction for Request msg REGISTER/cseq=5 (rdata0x7f400024f8)
[Dec 15 18:42:05] DEBUG[495]: res_pjsip/pjsip_distributor.c:471 ast_sip_get_distributor_serializer: Calculated serializer pjsip/distributor-00000032 to use for Request msg REGISTER/cseq=5 (rdata0x7f400024f8)
[Dec 15 18:42:05] DEBUG[567]: res_pjsip_endpoint_identifier_ip.c:275 common_identify: No identify sections to match against
[Dec 15 18:42:05] DEBUG[567]: res_pjsip_endpoint_identifier_user.c:148 username_identify: Attempting identify by From username '8001' domain '192.168.1.250'
[Dec 15 18:42:05] DEBUG[567]: res_pjsip_endpoint_identifier_user.c:152 username_identify: Endpoint not found for From username '8001' domain '192.168.1.250'
[Dec 15 18:42:05] DEBUG[567]: res_pjsip_endpoint_identifier_ip.c:275 common_identify: No identify sections to match against
[Dec 15 18:42:05] NOTICE[567]: res_pjsip/pjsip_distributor.c:676 log_failed_request: Request 'REGISTER' from '<sip:8001@192.168.1.250>' failed for '192.168.1.110:5060' (callid: 156578329@192.168.1.110) - No matching endpoint found
[Dec 15 18:42:05] SECURITY[563]: res_security_log.c:114 security_event_stasis_cb: SecurityEvent="InvalidAccountID",EventTV="2022-12-15T18:42:05.821+0100",Severity="Error",Service="PJSIP",EventVersion="1",AccountID="8001",SessionID="156578329@192.168.1.110",LocalAddress="IPV4/UDP/192.168.1.250/5060",RemoteAddress="IPV4/UDP/192.168.1.110/5060"
[Dec 15 18:42:05] DEBUG[567]: res_pjsip_authenticator_digest.c:358 verify: Realm: asterisk  Username:   Result: NOAUTH
[Dec 15 18:42:05] SECURITY[563]: res_security_log.c:114 security_event_stasis_cb: SecurityEvent="ChallengeSent",EventTV="2022-12-1
![Screenshot_20221215_184758_Samsung Internet](https://user-images.githubusercontent.com/76753636/207932413-d0382489-54c5-454d-83de-0f3aedbaa8c3.png)
5T18:42:05.821+0100",Severity="Informational",Service="PJSIP",EventVersion="1",AccountID="<unknown>",SessionID="156578329@192.168.1.110",LocalAddress="IPV4/UDP/192.168.1.250/5060",RemoteAddress="IPV4/UDP/192.168.1.110/5060",Challenge=""

**Comment by pilurizzu** [comment]
Dahua 3221

**Comment by TECH7Fox** [comment]
Some of these settings are for chan_sip only, not pjsip. Here is a pjsip example:
```
[008]
type = endpoint
context = default
disallow = all
allow = alaw
allow = ulaw
direct_media_method=invite
callerid="jordy" <008>
dtmf_mode=inband
force_rport=no
aors= 008
auth = auth008

[008]
type = aor
max_contacts = 1

[auth008]
type=auth
auth_type=userpass
password=1234
username=008
```
